### ISSUE TITLE
[BUG] `prefix:` not working correctly. (was: StorageException with fresh installation on Mysql)

### ISSUE THREAD

**Issue by nbehier** [bolt/bolt#5385]
I try to install Bolt 3. First, I untar the last archive (3.0.1) then load bolt on my browser and get `index.php/bolt/userfirst` page. As the `config.yml` file is created, I change the database section to :

``` (yml)
database:
    driver: mysql
    databasename: xxx
    username: xxx
    password: xxx
    host: xxx
    prefix: test_b3_
```

When I reload the `index.php/bolt/userfirst` page, I get `StorageException` :

``` (php)
Error: Uncaught exception 'Bolt\Exception\StorageException' with message 'Attempted to load mapping data for unmapped class Bolt\Storage\Entity\Users' in /vendor/bolt/bolt/src/Storage/Mapping/MetadataDriver.php:371

Stack trace:
#0 /vendor/bolt/bolt/src/Storage/EntityManager.php(256): Bolt\Storage\Mapping\MetadataDriver->loadMetadataForClass('Bolt\\Storage\\En...')
#1 /vendor/bolt/bolt/src/Users.php(40): Bolt\Storage\EntityManager->getRepository('Bolt\\Storage\\En...')
#2 /vendor/bolt/bolt/src/Provider/UsersServiceProvider.php(15): Bolt\Users->__construct(Object(Bolt\Application))
#3 /vendor/pimple/pimple/lib/Pimple.php(126): Bolt\Provider\UsersServiceProvider->Bolt\Provider\{closure}(Object(Bolt\Application))
#4 /vendor/pimple/pimple/lib/Pimple.php(83): Bolt\Application::{closure}(Object(Bolt\Application))
#5 /vendor/bolt/bolt/src/Provider/TwigServiceProvider.php(83): Pim

File: /vendor/bolt/bolt/src/Storage/Mapping/MetadataDriver.php
Line: 371
```
## Details
- Relevant Bolt Version: 3.0.1
- Install type: Zip/tarball
- PHP version: 5.6
- Used webserver: Apache

**Comment by ychen8610** [comment]
it happen when db driver is set to mysql. any update for this bug?

**Comment by GawainLynch** [comment]
My guess on the reported bug is the use of `prefix:` **might** be causing a collision.

**Comment by nbehier** [comment]
In my Bolt v2, I used `prefix`, is there another way to specify my tables prefix in v3 ?

**Comment by GawainLynch** [comment]
It *should* work, I am just isolating what "stands out" to me that might be worth trying

**Comment by SahAssar** [comment]
Since I don't have a mysql install on hand right now I'll just ask here: Did it ever get confirmed that is an issue with `prefix`?

**Comment by nbehier** [comment]
On a bolt v3.0.3, if I change `database` from `sqlite` to `mysql` with `prefix` (with or without underscore on it), I get the Exception. I get the Exception from `php app/nut cache:clear` too. I'll try to test on last version of Bolt to see if happen again.

**Comment by nbehier** [comment]
I test on v3.0.11, exactly the same. 

1. Declare `sqlite` as database provider on my `config.yml`
2. `php app/nut database:check` => correct
3. Update my Bolt with tarball
4. `php app/nut cache:clear`
5. Declare `mysql` as database provider with a `prefix` property on my `config.yml`
6. `php app/nut cache:clear` or `php app/nut database:check` and get the Exception.

=> I could not easily at this time recheck with `mysql` without `prefix` because I already have a Bolt production site on this database.

**Comment by nbehier** [comment]
OK, I checked all the same as before, on a new database, but without `prefix` instruction on `config.yml` => perfectly works.

When I checked on `bolt/bolt/src/Storage/Mapping/MetadataDriver.php`, I do not see any prefix code… [comments on line 94](https://github.com/bolt/bolt/blob/release/3.0/src/Storage/Mapping/MetadataDriver.php#L94) seems to be clear on that point.

**Comment by SahAssar** [comment]
Assigning @rossriley to this since it that seems to be the area (to my untrained eye)...

**Comment by nbehier** [comment]
Hi, do you plan to resolve this prefix bug soon ?
It would be so great and enable me to have several bolt websites on the same MySQL database :-)

**Comment by nbehier** [comment]
@rossriley @GawainLynch I try again with fresh install. Both Bolt 3.1.5 and 3.2.0-beta 5 have ever this error. If I try without `prefix`, I get First User creation page, but with `prefix`option, I get an error
```
Uncaught Exception: InvalidRepositoryException

InvalidRepositoryException in EntityManager.php line 259: 
Attempted to load repository for invalid class or alias: Bolt\Storage\Entity\Users. Check that the class, alias or contenttype definition is correct.
```

Stack Trace begins with
```
# 1	Bolt\Storage\EntityManager::getRepository("Bolt\Storage\Entity\Users") 
[root]/vendor/bolt/bolt/src/Users.php # line 45
# 2	Bolt\Users::getRepository() 
[root]/vendor/bolt/bolt/src/Users.php # line 185
# 3	Bolt\Users::getUsers() 
[root]/vendor/bolt/bolt/src/Controller/Base.php # line 274
# 4	Bolt\Controller\Base::hasUsers() 
[root]/vendor/bolt/bolt/src/Controller/Frontend.php # line 56
# 5	Bolt\Controller\Frontend::before(Request, Application) 
# 6	call_user_func([array], Request, Application) 
[root]/vendor/bolt/bolt/src/Controller/ConfigurableBase.php # line 124
```

**Comment by GawainLynch** [comment]
@SahAssar @nbehier confirmed non-fix here … reopening.

**Comment by GawainLynch** [comment]
Bonus points to the person who spots the :koala: here … and not the one in the prefix name:
![image](https://cloud.githubusercontent.com/assets/1427081/19474956/1b8ca85a-9532-11e6-8b29-ba6df21b7466.png)

**Comment by bobdenotter** [comment]
@GawainLynch I'm probably dense, but I don't spot the problem. Are you saying this is an easy fix?

**Comment by GawainLynch** [comment]
Not sure about ease, and yet more changed around that stuff in the meantime.

**Comment by GawainLynch** [comment]
Fixed in #6300
